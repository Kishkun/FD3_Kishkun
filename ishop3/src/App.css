*:focus {
  box-sizing: border-box;
  margin: 0;
  outline: none
}
a {
  outline: none;
  text-decoration: none;
  color: #2B2A29;
}
li {
  list-style-type: none;
}
ul {
  padding: 0;
}
html,
body {
  padding: 0;
  min-height: 100vh;
  font-family: 'Lato', sans-serif;
  font-size: 14px;
  max-width: 1920px;
  margin: 0 auto;
  color: #2B2A29;
  background-color: #ffffff;
  /*400 700 900*/
}
.shop_wrapper {
  max-width: 1200px;
  margin: 0 auto;
  padding-top: 60px;
}
.shop_wrapper .shop_title {
  text-align: center;
  font-size: 3vw;
  font-weight: 700;
  margin-bottom: 40px;
}
.mobile_bg {
  background: linear-gradient(122.04deg,#14e9b6 23.38%,#076deb 82.75%);
}