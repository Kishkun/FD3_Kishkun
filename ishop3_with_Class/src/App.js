import React from 'react';
import PropTypes from 'prop-types';
import Mobile from './components/mobile/Mobile';
import Header from './components/header/Header';
import Footer from './components/footer/Footer';

class App extends React.Component {

    state = {
        mobileList: this.props.mobiles,
    };

    deleteItem = (id) => {
        let filteredList = this.state.mobileList.filter(item => {
            return item.id !== id
        });
        this.setState({mobileList: filteredList});
        console.log(id);
    };

    render() {
        return (
            <div className="shop_wrapper">
                <Header title={this.props.title}/>
                <main>
                    <Mobile mobiles={this.state.mobileList}
                            deleteItem={this.deleteItem}
                    />
                </main>
                <Footer/>
            </div>
        );
    }
}

App.defaultProps = {
    title: 'Online store',
    mobiles: []
};

App.propTypes = {
    title: PropTypes.string.isRequired,
    mobiles: PropTypes.array.isRequired,
};

export default App;
