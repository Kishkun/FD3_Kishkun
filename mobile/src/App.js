import React from 'react';
import PropTypes from 'prop-types';
import myEvents from './events';
import CompanyName from './components/CompanyName';
import MobileClients from './components/clients/MobileClients';
import AddedClientButton from './components/buttons/AddedClientButton';
import FilterStatusButtons from './components/buttons/FilterStatusButtons';
import EditClientItem from './components/options/EditClientItem';
import AddClientItem from './components/options/AddClientItem';

class App extends React.PureComponent {

    static propTypes = {
        title: PropTypes.string.isRequired,
        companies: PropTypes.arrayOf(
            PropTypes.shape({
                id: PropTypes.number.isRequired,
                name: PropTypes.string.isRequired
            })
        ),
        clients:PropTypes.arrayOf(
            PropTypes.shape({
                id: PropTypes.number.isRequired,
                firstName: PropTypes.string.isRequired,
                secondName: PropTypes.string.isRequired,
                patronymic: PropTypes.string.isRequired,
                balance: PropTypes.number.isRequired,
                status: PropTypes.bool.isRequired
            })
        ),
    };

    constructor(props) {
        super(props);
        this.state = {
            filter: 'all', //active, all, blocked,
            clients: this.props.clients,
            companies: this.props.companies,
            title: this.props.title,
            editingItem: null,
            isShowAddedForm: false
        };
        this.minID = this.props.clients.length;
        myEvents.on('onFilterChange', this.onFilterChange.bind(this));
        myEvents.on('onDeleteItem', this.onDeleteItem.bind(this));
        myEvents.on('onEditItem', this.onEditItem.bind(this));
        myEvents.on('closeForm', this.closeForm.bind(this));
        myEvents.on('onClientChange', this.onClientChange.bind(this));
        myEvents.on('onOpenAddedItem', this.onOpenAddedItem.bind(this));
        myEvents.on('onAddItem', this.onAddItem.bind(this));
    }

    minID = 1;

    createId = (item) => {
        return {
            id: ++this.minID,
            ...item
        };
    };

    filter(clients, filter) {
        switch (filter) {
            case 'all':
                return clients;
            case 'active':
                return clients.filter((item) => item.status);
            case 'blocked':
                return clients.filter((item) => !item.status);
            default:
                return clients;
        }
    }

    onDeleteItem = (id) => {
        let filteredList = this.state.clients.filter(item => item.id !== id);
        this.setState({clients: filteredList});
    };

    onFilterChange = (filter) => {
        this.setState({filter: filter})
    };

    onEditItem = (id) => {
        let editingItem = this.state.clients.find(item => item.id === id);
        this.setState({editingItem: editingItem});
    };

    onClientChange = (client) => {
        this.setState((state) =>{
            let findClient = state.clients.findIndex((el) => el.id === client.id);
            let newClients = [...state.clients];
            newClients[findClient] = client;
            return {
                clients: newClients,
                editingItem: null
            }
        });
    };

    onOpenAddedItem = () => {
        this.setState({isShowAddedForm: true});
    };

    closeForm = () =>{
        this.setState({editingItem: null, isShowAddedForm: false});
    };

    onAddItem = (item) => {
        let newItem = this.createId(item);
        let newList = [...this.state.clients, newItem];
        this.setState({
            clients: newList,
            isShowAddedForm: false,
        });
    };

    render() {
        console.log("MobileCompany render");
        let {filter, clients, companies, title} = this.state;
        let filtered = this.filter(clients, filter);
        return (
            <div className='company_wrapper'>
                <h1 className='company_title'>{title}</h1>
                <CompanyName companies={companies} />
                <hr/>
                <FilterStatusButtons filter={filter} />
                <hr/>
                <MobileClients
                    clients={filtered}/>
                <hr/>
                <AddedClientButton/>
                {this.state.isShowAddedForm ? <AddClientItem/> : false}
                {this.state.editingItem ? <EditClientItem
                    key={this.state.editingItem.id}
                    editingItem={this.state.editingItem} /> : null}
            </div>
        );
    }
}

export default App;
