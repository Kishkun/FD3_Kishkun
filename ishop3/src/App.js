import React from 'react';
import PropTypes from 'prop-types';
import './App.css';
import Mobile from './components/mobile/Mobile';

let App = (props) => {
    return (
        <div className="shop_wrapper">
            <header>
                <h1 className='shop_title'>{props.title}</h1>
            </header>
            <Mobile mobiles={props.mobiles}
                    removeItem={props.removeItem}
                    toggleBackgroundItem={props.toggleBackgroundItem}
            />
        </div>
    );
};

App.defaultProps = {
    title: 'Online store',
    mobiles: []
};

App.propTypes = {
    title: PropTypes.string.isRequired,
    mobiles: PropTypes.array.isRequired,
};

export default App;
