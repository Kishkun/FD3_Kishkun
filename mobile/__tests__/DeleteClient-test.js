import {mount, shallow} from 'enzyme';
import React from 'react';
import App from '../../mobile/src/App';

test('тестирование редактирование клиента', () => {

    let clients = [
        {
            'id': 1,
            'firstName': 'Иванов',
            'secondName': 'Иван',
            'patronymic': 'Иванович',
            'balance': 200,
            'status': true
        }
    ];
    let title = 'Test App';
    let companies = [
        {
            'id': 1,
            'name': 'Velcom'
        }
    ];

    const wrapper = mount(
        <App clients={clients}
             title={title}
             companies={companies}/>
    );

    expect(wrapper).toMatchSnapshot();

    let deleteButton = wrapper.find('.delete_btn');
    deleteButton.simulate('click');
    expect(wrapper).toMatchSnapshot();
});