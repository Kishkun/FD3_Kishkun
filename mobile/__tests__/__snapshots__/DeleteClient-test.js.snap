// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`тестирование редактирование клиента 1`] = `
<App
  clients={
    Array [
      Object {
        "balance": 200,
        "firstName": "Иванов",
        "id": 1,
        "patronymic": "Иванович",
        "secondName": "Иван",
        "status": true,
      },
    ]
  }
  companies={
    Array [
      Object {
        "id": 1,
        "name": "Velcom",
      },
    ]
  }
  title="Test App"
>
  <div
    className="company_wrapper"
  >
    <h1
      className="company_title"
    >
      Test App
    </h1>
    <CompanyName
      companies={
        Array [
          Object {
            "id": 1,
            "name": "Velcom",
          },
        ]
      }
    >
      <div
        className="name_block"
      >
        <CompanyNameButton
          activeCompanyId={
            Object {
              "id": 1,
              "name": "Velcom",
            }
          }
          item={
            Object {
              "id": 1,
              "name": "Velcom",
            }
          }
          key="1"
        >
          <button
            className="name_btn active"
            key="1"
            onClick={[Function]}
            type="button"
          >
            Velcom
          </button>
        </CompanyNameButton>
        <p
          className="name"
        >
          Компания: 
          Velcom
        </p>
      </div>
    </CompanyName>
    <hr />
    <FilterStatusButtons
      filter="all"
    >
      <div
        className="filter_block"
      >
        <button
          className="clients_btn active"
          data-name="all"
          key="all"
          onClick={[Function]}
          type="button"
        >
          Все
        </button>
        <button
          className="clients_btn"
          data-name="active"
          key="active"
          onClick={[Function]}
          type="button"
        >
          Активные
        </button>
        <button
          className="clients_btn"
          data-name="blocked"
          key="blocked"
          onClick={[Function]}
          type="button"
        >
          Заблокированые
        </button>
      </div>
    </FilterStatusButtons>
    <hr />
    <MobileClients
      clients={
        Array [
          Object {
            "balance": 200,
            "firstName": "Иванов",
            "id": 1,
            "patronymic": "Иванович",
            "secondName": "Иван",
            "status": true,
          },
        ]
      }
      isShowAddedForm={false}
    >
      <div
        className="clients_wrapper"
      >
        <div
          className="clients_title"
        >
          <div
            className="th"
          >
            Фамилия
          </div>
          <div
            className="th"
          >
            Имя
          </div>
          <div
            className="th"
          >
            Отчество
          </div>
          <div
            className="th"
          >
            Баланс
          </div>
          <div
            className="th"
          >
            Статус
          </div>
          <div
            className="th button_th"
          >
            Редактировать
          </div>
          <div
            className="th button_th"
          >
            Удалить
          </div>
        </div>
        <ClientItem
          isShowAddedForm={false}
          item={
            Object {
              "balance": 200,
              "firstName": "Иванов",
              "id": 1,
              "patronymic": "Иванович",
              "secondName": "Иван",
              "status": true,
            }
          }
          key="1"
        >
          <div
            className="client_block"
          >
            <div
              className="td firstName"
            >
              Иванов
            </div>
            <div
              className="td secondName"
            >
              Иван
            </div>
            <div
              className="td patronymic"
            >
              Иванович
            </div>
            <div
              className="td balance"
            >
              200
            </div>
            <div
              className="td status active"
            >
              active
            </div>
            <div
              className="td button_td"
            >
              <button
                className="clients_btn edit_btn"
                disabled={false}
                onClick={[Function]}
                type="button"
              >
                Редактировать
              </button>
            </div>
            <div
              className="td button_td"
            >
              <button
                className="clients_btn delete_btn"
                disabled={false}
                onClick={[Function]}
                type="button"
              >
                Удалить
              </button>
            </div>
          </div>
        </ClientItem>
      </div>
    </MobileClients>
    <hr />
    <AddedClientButton>
      <button
        className="clients_btn add_btn"
        onClick={[Function]}
        type="button"
      >
        Добавить клиента
      </button>
    </AddedClientButton>
  </div>
</App>
`;

exports[`тестирование редактирование клиента 2`] = `
<App
  clients={
    Array [
      Object {
        "balance": 200,
        "firstName": "Иванов",
        "id": 1,
        "patronymic": "Иванович",
        "secondName": "Иван",
        "status": true,
      },
    ]
  }
  companies={
    Array [
      Object {
        "id": 1,
        "name": "Velcom",
      },
    ]
  }
  title="Test App"
>
  <div
    className="company_wrapper"
  >
    <h1
      className="company_title"
    >
      Test App
    </h1>
    <CompanyName
      companies={
        Array [
          Object {
            "id": 1,
            "name": "Velcom",
          },
        ]
      }
    >
      <div
        className="name_block"
      >
        <CompanyNameButton
          activeCompanyId={
            Object {
              "id": 1,
              "name": "Velcom",
            }
          }
          item={
            Object {
              "id": 1,
              "name": "Velcom",
            }
          }
          key="1"
        >
          <button
            className="name_btn active"
            key="1"
            onClick={[Function]}
            type="button"
          >
            Velcom
          </button>
        </CompanyNameButton>
        <p
          className="name"
        >
          Компания: 
          Velcom
        </p>
      </div>
    </CompanyName>
    <hr />
    <FilterStatusButtons
      filter="all"
    >
      <div
        className="filter_block"
      >
        <button
          className="clients_btn active"
          data-name="all"
          key="all"
          onClick={[Function]}
          type="button"
        >
          Все
        </button>
        <button
          className="clients_btn"
          data-name="active"
          key="active"
          onClick={[Function]}
          type="button"
        >
          Активные
        </button>
        <button
          className="clients_btn"
          data-name="blocked"
          key="blocked"
          onClick={[Function]}
          type="button"
        >
          Заблокированые
        </button>
      </div>
    </FilterStatusButtons>
    <hr />
    <MobileClients
      clients={Array []}
      isShowAddedForm={false}
    >
      <div
        className="clients_wrapper"
      >
        <div
          className="clients_title"
        >
          <div
            className="th"
          >
            Фамилия
          </div>
          <div
            className="th"
          >
            Имя
          </div>
          <div
            className="th"
          >
            Отчество
          </div>
          <div
            className="th"
          >
            Баланс
          </div>
          <div
            className="th"
          >
            Статус
          </div>
          <div
            className="th button_th"
          >
            Редактировать
          </div>
          <div
            className="th button_th"
          >
            Удалить
          </div>
        </div>
      </div>
    </MobileClients>
    <hr />
    <AddedClientButton>
      <button
        className="clients_btn add_btn"
        onClick={[Function]}
        type="button"
      >
        Добавить клиента
      </button>
    </AddedClientButton>
  </div>
</App>
`;
